### DISCUSSION TITLE
Cleaning Libraries and Code

### DISCUSSION THREAD

**Discussion by poolborges** [NOSiCode-CV/IGRP-Framework#1754] [Ideas]
We need to clean the code without add breaking changes, for future modullarity, better code test

* igrp core-framework
* igrpweb

### 1. Remove JDK libraries

- [tools.jar](https://github.com/NOSiCode-CV/IGRP-Framework/blob/master/IGRP/src/main/webapp/WEB-INF/lib/tools.jar)
- Image size: ~18MB

**Why the library is here ?**

"Studio" use sun compiler to compile generated "java" code

**Solution 1** Common Practices

Should have requirement for JDK not JRE.
- Is common practices when we install any Servlet Container(Tomcar, Jetty) or application server (Wildfly, ...)

**Solution 2** (Future/IDEIA): Introducing "Builder"

* [Paulo Borges igrp-experimental](https://github.com/poolborges/igrp-experimental)/[docs](https://github.com/poolborges/igrp-experimental/tree/main/docs/IGRP-Builder.adoc)

### 2. Remove CXF and use of Jersey as JAX-RS Client

Should not use two library for same SPEC

* JAX-RS Spec don't provide API for "multipart/form-data", 
* Jersey provide "personal" implementation for this so there is no need to add CXF (with a lot of dependencies just to solve this)

**Solution 1** Remove Apache CXF

.Apache CXF 
`
<cxf.rt.frontend.jaxrs.version>3.3.6</cxf.rt.frontend.jaxrs.version>
<tomee.jaxrs.version>8.0.4</tomee.jaxrs.version>
`

.Jersey (Provide everything for form-multipart)
`
<jersey.version>2.29</jersey.version>
`
**Solution 2** Better API to integrate with BPMN repository server

This MUST support the Activity 6.0.0 and also allow planning support for other: Activity 7.0.0, ...

### 3 Move Embedded Tomee from core

Should NOT implement Embedded Server inside core framework
Should be implemented as another maven artifact ("igrpweb-server-tomee)
BETTER to use springboot ("igrpweb-server-spboot)
-- IGRP is has a inhouse implement of "Front controller(FC) pattern "
-- Spring-Web has a powerful FC implementation .
--- How (S1): Register everything insise (web.xml) to spring and let spring do the magic

.Maven 
`
<tomee.embedded.version>8.0.7</tomee.embedded.version>
`

**Comment by NOSiopen** [comment]
Hello @poolborges 
We have made come changes that you suggested, updated alot of things and we have moved do maven https://mvnrepository.com/artifact/cv.igrp like you can see in the readme.

Please review this new version and give your precious inputs

**Comment by kedargithub-stack** [comment]
Is there any framework KT session is available.

**Reply by NOSiopen** [reply]
We are working now in IGRP 3.0 and we want to make those sessions. 
Stay tuned in [igrp.cv](https://igrp.cv/en) and will report sessions
